### PR TITLE
[FLINK-30727][table-planner] Fix JoinReorderITCaseBase.testBushyTreeJoinReorder failed

### PR DESCRIPTION
**Pull request by swuferhong**
## What is the purpose of the change

This pr is aims to fix JoinReorderITCaseBase.testBushyTreeJoinReorder failed while running CI.  The failed CI link: [https://dev.azure.com/apache-flink/apache-flink/_build/results?buildId=44985&view=logs&j=7e1b167a-ddfd-53ed-4835-e68a01b608a7&t=434b1f2b-ca64-5a0b-f3a5-4a191160d4cb](url)

## Verifying this change

No tests.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive):  no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector:  no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? no docs

### CODE REVIEW

File: flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/runtime/stream/sql/join/JoinReorderITCase.java [hunk 1]
```diff
@@ -42,6 +45,12 @@ public class JoinReorderITCase extends JoinReorderITCaseBase {
 
     private StreamExecutionEnvironment env;
 
+    @AfterEach
+    public void after() {
```
Comment (lsyldliu) [inline flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/runtime/stream/sql/join/JoinReorderITCase.java:49]: This is the root cause? Do you reproduce failed test locally?

**Comment by swuferhong** [comment]
@flinkbot run azure

File: flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/runtime/stream/sql/join/JoinReorderITCase.java (hunk 1 above)
Comment (swuferhong) [reply flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/runtime/stream/sql/join/JoinReorderITCase.java:49]: > This is the root cause? Do you reproduce failed test locally?

This error can not be reproduce in local. I think the root cause is the `StreamTestSink.clear();` not be called after tests finished.

**Comment by swuferhong** [comment]
@flinkbot run azure

**Comment by swuferhong** [comment]
@flinkbot run azure

**Comment by MartijnVisser** [comment]
@swuferhong @lsyldliu What's the status of this PR? This test is a release blocker so we would like to resolve it asap

**Comment by swuferhong** [comment]
> @swuferhong @lsyldliu What's the status of this PR? This test is a release blocker so we would like to resolve it asap

Hi, @MartijnVisser. Now, I located that this error may be related to Flink runtime, and it is under intensive processing. (@TanYuxin-tyx will help me to fix it).

**Comment by swuferhong** [comment]
@flinkbot run azure

**Comment by swuferhong** [comment]
@flinkbot run azure

**Comment by swuferhong** [comment]
> @flinkbot run azure

The result of this round is: SUCCESS

**Comment by swuferhong** [comment]
@flinkbot run azure

**Comment by swuferhong** [comment]
@flinkbot run azure

**Comment by swuferhong** [comment]
@flinkbot run azure

**Comment by swuferhong** [comment]
@flinkbot run azure

**Comment by swuferhong** [comment]
@flinkbot run azure

**Comment by swuferhong** [comment]
@flinkbot run azure

File: flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/runtime/stream/sql/join/JoinReorderITCase.java [hunk 2]
```diff
@@ -42,6 +47,20 @@ public class JoinReorderITCase extends JoinReorderITCaseBase {
 
     private StreamExecutionEnvironment env;
 
+    @BeforeEach
+    public void before() throws Exception {
+        super.before();
+        // This conf is aims to fix IOException due to timeout occurring while requesting exclusive
+        // NetworkBuffer (see https://issues.apache.org/jira/browse/FLINK-30727).
+        tEnv.getConfig().getConfiguration().set(TaskManagerOptions.NETWORK_MEMORY_FRACTION, 0.4f);
```
Comment (godfreyhe) [inline flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/runtime/stream/sql/join/JoinReorderITCase.java:55]: make sense to me this fix can make the test stable, but eventually we should remove this config

**Review by godfreyhe — APPROVED** [review]
LGTM

**Pull request closed without merging**